src/main/java/com/example/document/exceptionHandler/DocumentExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package com.example.document.exceptionHandler;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestControllerAdvice
+public class DocumentExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(value = {HttpClientErrorException.class})
+    protected ResponseEntity<Object> handleConflict(RuntimeException ex, WebRequest request) {
+
+        String errorMessage = "\"Document upload failed!\"\n" + ex.getMessage().split(":")[1].strip();
+
+        return handleExceptionInternal(ex, errorMessage,
```

Comment:
Why do you want to return code when having request when employee is missing in Employee DB?

Reply:
to inform the user that the operation is suspended

Reply:
Let's catch not found exception from Employee Service in `EmployeeClient` and throw some specific exception instead of handling all issues with this handler